### PR TITLE
fix(oci/pusher): minor fixes in the pusher logic

### PR DESCRIPTION
**Pull request by alacuku**
Signed-off-by: Aldo Lacuku <aldo@lacuku.eu>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

> /kind flaky-test

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area library

> /area cli

/area tests

> /area examples

**What this PR does / why we need it**:
This PR introduces a bunch of bug fixes in the pusher logic and adds tests:
* fix panic in pusher when the platform is not in the right format;
* always use "index image" when pushing plugin artifacts: fixes the missing platform info when only one plugin is pushed;
* only push one rulesfile artifact and error when multiple filepaths are passed: fixes the case where an image index is pushed referencing mutliple rulesfiles.
* error when dependencies are passed for plugin artifacts;
* add tests for the pusher logic: 100% file and 89.7% statements coverage for the pusher package.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: cmd/registry_push.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ func (o *pushOptions) RunPush(ctx context.Context, args []string) error {
 
 	client := authn.NewClient(cred)
 
-	pusher := ocipusher.NewPusher(client, newPushProgressTracker(o.Printer))
+	pusher := ocipusher.NewPusher(client, false, newPushProgressTracker(o.Printer))
```
Comment (loresuso) [inline cmd/registry_push.go:130]: Let's remember to have a flag for PlainHTTP, but I agree that this is not the purpose of this PR

**Review by loresuso — COMMENTED** [review]
/lgtm

File: pkg/oci/pusher/pusher.go [hunk 2]
```diff
@@ -72,11 +86,22 @@ func (p *Pusher) Push(ctx context.Context, artifactType oci.ArtifactType,
 		return nil, err
 	}
 
+	// First thing check that we do not have multiple rulesfiles.
+	if artifactType == oci.Rulesfile && len(o.Filepaths) != 1 {
```
Comment (FedeDP) [inline pkg/oci/pusher/pusher.go:90]: I would love if artifactType was used to build an `Artifact` interface; we could use here
`artifact.Validate()`
and below
`artifact.RootDesc()`

Wdyt?

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**